.github/workflows/update_documentation.yml

```diff
@@ -0,0 +1,92 @@
+name: "Upload documentation to website"
+
+description: "This action updates the documentation on the t8code website"
+
+on:
+  push:
+    branches:
+      - main
+      - feature-automate_documentation_update_over_CI
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    
+    timeout-minutes: 20
+    steps:
+#
+# Setup and bootstrap
+#
+    - uses: actions/checkout@v1
+    - name: Install mpich
+      run: sudo apt-get update && sudo apt-get install libmpich-dev
+    - name: Install zlib
+      run: sudo apt-get install libz-dev
+    - name: init submodules
+      run: git submodule init
+    - name: update submodules
+      run: git submodule update
+    - name: bootstrap
+      run: ./bootstrap
+
+    # build config vars
+    - name: build CFLAGS and CXXFLAGS variables
+      run: echo CFLAGS_var="-Wall -pedantic -O3" >> $GITHUB_ENV
+        && echo CXXFLAGS_var="-Wall -pedantic -O3" >> $GITHUB_ENV
+    # Note: We want to use '-Werror', but if we already provide it at the configure step
+    #       we get errors in configure that we cannot remove (for example the autotools way
+    #       of checking for libm results in a compiler warning).
+    #       Thus, we add '-Werror' later at the make step.
+    - name: build config variables
+      run: export CONFIG_OPTIONS="--without-blas ${LESS_TEST_OPTION}"
+           && export CONFIG_SERIAL_DEBUG="$CONFIG_OPTIONS --enable-debug --with-sc=$SC_SERIAL_DEBUG/install --with-p4est=$P4EST_SERIAL_DEBUG/install"
+           && echo CONFIG_OPTIONS="$CONFIG_OPTIONS" >> $GITHUB_ENV
+           && echo CONFIG_SERIAL_DEBUG="$CONFIG_SERIAL_DEBUG" >> $GITHUB_ENV
+    - name: Check vars
+      run: echo "[$CONFIG_SERIAL_DEBUG]"
+# configure and generate doxygen
+    - name: check debugging mode
+      run: echo "Checking debug mode"
+    - name: configure
+      run: mkdir build_debug && cd build_debug && ../configure $CONFIG_DEBUG CFLAGS="$CFLAGS_var" CXXFLAGS="$CXXFLAGS_var"
+    - name: OnFailUploadLog
+      if: failure()
+      uses: actions/upload-artifact@v1
+      with:
+        name: config_debug_MPI.log
+        path: build_debug/config.log
+    - name: make
+      run: cd build_debug && make -j V=0 CFLAGS="$CFLAGS_var -Werror" CXXFLAGS="$CXXFLAGS_var -Werror"
+    - name: make check
```

Comment:
This workflow should not check the implementation in my opinion. But it should only run, when all tests have passed. To ensure it, we can use the needs keyword, creating a dependancy between workflows: https://docs.github.com/en/actions/using-workflows/about-workflows#creating-dependent-jobs

Reply:
My bad, what I actually ment was: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run

Reply:
The workflow is meant to sync the documentation to the actual state of the main branch. Therefore, it is irrelevant, if the tests pass.

Reply:
But we shouldn't update a documentation of a crashing version.